### ISSUE TITLE
"log_loss" metric in binary problems shows "RuntimeError: expected scalar type Long but found Int" [BUG]

### ISSUE THREAD

**Issue by arturdaraujo** [autogluon/autogluon#1815]
- [x] I have checked that this bug exists on the latest stable version of AutoGluon
- [x] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**
For "log_loss" metric in binary problems, the `RuntimeError: expected scalar type Long but found Int` error occurs. I also tried with presets='medium_quality', 'high_quality', 'best_quality'. None of them worked.

**Expected behavior**
The code trains all the models, in the end just before finishing it gives the error.

**To Reproduce**
```
predictor = TabularPredictor(
    label=target_name,
    eval_metric='log_loss',
).fit(
    train,
    time_limit=60 * 2,
    verbosity=2,
    presets="medium_quality"
)
[...]
Fitting model: RandomForestEntr_BAG_L1_FULL | Skipping fit via cloning parent ...
	2.95s	 = Training   runtime
	0.39s	 = Validation runtime
Fitting 1 L1 models ...
Fitting model: CatBoost_BAG_L1_FULL ...
	2.36s	 = Training   runtime
Fitting model: WeightedEnsemble_L2_FULL | Skipping fit via cloning parent ...
	0.47s	 = Training   runtime
Traceback (most recent call last):

  File "C:\Users\artur\AppData\Local\Temp\ipykernel_3116\4285418363.py", line 1, in <cell line: 1>
    predictor = TabularPredictor(

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\autogluon\core\utils\decorators.py", line 30, in _call
    return f(*gargs, **gkwargs)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\autogluon\tabular\predictor\predictor.py", line 823, in fit
    self._post_fit(

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\autogluon\tabular\predictor\predictor.py", line 872, in _post_fit
    self._calibrate_model()

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\autogluon\tabular\predictor\predictor.py", line 942, in _calibrate_model
    temp_scalar = tune_temperature_scaling(y_val_probs=y_val_probs, y_val=y_val,

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\autogluon\core\calibrate\temperature_scaling.py", line 57, in tune_temperature_scaling
    optimizer.step(temperature_scale_step)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\optim\lr_scheduler.py", line 65, in wrapper
    return wrapped(*args, **kwargs)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\optim\optimizer.py", line 88, in wrapper
    return func(*args, **kwargs)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\autograd\grad_mode.py", line 28, in decorate_context
    return func(*args, **kwargs)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\optim\lbfgs.py", line 311, in step
    orig_loss = closure()

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\autograd\grad_mode.py", line 28, in decorate_context
    return func(*args, **kwargs)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\autogluon\core\calibrate\temperature_scaling.py", line 52, in temperature_scale_step
    loss = nll_criterion(new_logits, y_val_tensor)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\nn\modules\module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\nn\modules\loss.py", line 1150, in forward
    return F.cross_entropy(input, target, weight=self.weight,

  File "C:\Users\artur\miniconda3\envs\cf_predict38\lib\site-packages\torch\nn\functional.py", line 2846, in cross_entropy
    return torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing)

RuntimeError: expected scalar type Long but found Int

```

**Installed Versions**
Which version of AutoGluon are you are using?  

```python
# Replace this code with the output of the following:
from autogluon.core.utils import show_versions
show_versions()
```

INSTALLED VERSIONS
------------------
date                 : 2022-06-13
time                 : 23:30:34.858671
python               : 3.9.13.final.0
OS                   : Windows
OS-release           : 10
Version              : 10.0.22000
machine              : AMD64
processor            : Intel64 Family 6 Model 142 Stepping 12, GenuineIntel
num_cores            : 8
cpu_ram_mb           : 16197
cuda version         : None
num_gpus             : 0
gpu_ram_mb           : [2010]
avail_disk_size_mb   : None

autogluon.common     : 0.4.2
autogluon.core       : 0.4.2
autogluon.features   : 0.4.2
autogluon.tabular    : 0.4.2
autogluon.text       : 0.4.2
autogluon.vision     : 0.4.2
autogluon_contrib_nlp: None
boto3                : 1.24.3
catboost             : 1.0.6
dask                 : 2021.11.2
distributed          : 2021.11.2
fairscale            : 0.4.6
fastai               : 2.5.6
gluoncv              : 0.11.0
lightgbm             : 3.3.2
matplotlib           : 3.5.2
networkx             : 2.8.3
nptyping             : 1.4.4
numpy                : 1.22.4
omegaconf            : 2.1.2
pandas               : 1.3.5
PIL                  : 9.0.1
protobuf             : None
psutil               : 5.8.0
pytorch_lightning    : 1.6.4
ray                  : 1.10.0
requests             : 2.27.1
scipy                : 1.7.3
sentencepiece        : None
setuptools           : 59.5.0
skimage              : 0.19.2
sklearn              : 1.0.2
smart_open           : 5.2.1
timm                 : 0.5.4
torch                : 1.10.2+cpu
torchmetrics         : 0.7.3
tqdm                 : 4.64.0
transformers         : 4.16.2
xgboost              : 1.4.2

**Additional context**

**Comment by galtimur** [comment]
I have solve the problem by replacing the line of the code in the following file:

Anaconda\Lib\site-packages\autogluon\core\calibrate\temperature_scaling.py

line 33: y_val = y_val.astype('int')
new line: y_val = y_val.astype('int64')

**Comment by Innixma** [comment]
Thanks! This has been fixed in #2657 (v0.6.2 release)
